mac

```diff
@@ -166,29 +166,62 @@ brew_install_or_upgrade 'hub'
 # shellcheck disable=SC2016
 append_to_file "$HOME/.zshrc" 'eval "$(hub alias -s)"'
 
+
+fancy_echo 'Checking on Node.js installation...'
+
 if ! brew_is_installed "node"; then
-  if ! command -v n > /dev/null; then
-    fancy_echo 'Installing n and latest Node.js and NPM...'
-    curl -L http://git.io/n-install | bash -s -- -y latest
+  if command -v nvm > /dev/null; then
+    if ! command -v n > /dev/null; then
+      fancy_echo 'Installing n and latest Node.js and NPM...'
+      curl -L http://git.io/n-install | bash -s -- -y latest
+    else
+      fancy_echo 'Updating n...'
+      n-update -y
+    fi
   else
-    fancy_echo 'Updating n...'
-    n-update -y
+    fancy_echo 'nvm detected.  Skipping...'
   fi
+else
+  brew_install_or_upgrade 'node'
 fi
 
-brew_install_or_upgrade 'python3'
+fancy_echo '...Finished Node.js installation checks.'
+
+
+fancy_echo 'Checking on Python installation...'
+
+if ! brew_is_installed "python3"; then
```

Comment:
Hmmm, yes, you're right, though the goal here is not to install Python 3 directly if it wasn't already previously.  In other words, if a person runs this script for the first time, then the goal is to only install it through `pyenv`.  Otherwise, revert back to the original behavior from before so that they are guaranteed to get an update if one is available.  I was trying to mimic the updated Node.js installation behavior in regards to backwards compatibility of folks who have previously run the script.

Do you think it's overkill to do that, though?